src/project/internal/projectactionscontroller.cpp

```diff
@@ -375,10 +375,14 @@ bool ProjectActionsController::saveProject(const io::path_t& path)
         return false;
     }
 
-    // TODO(save-to-cloud):
-    // - check whether the score is *currently* public (might have been changed on .com)
-    //   (if yes, show warning)
-    // - save indeed to the cloud, generating MP3 as necessary
+    if (project->isCloudProject()) {
+        RetVal<CloudProjectInfo> info = saveProjectScenario()->askCloudLocation(project);
```

Comment:
I've become very reluctant to comment because apparently my understanding isn't 100% correct either, but I doubt about this: I thought that when the user just presses Ctrl+S in a score that has already been saved to the cloud, the score should just be saved without asking a new location (that should only happen when using "Save as...", or when saving a score for the first time). @bkunda @Tantacrul please check if you agree about this :)

Reply:
Yes, @cbjeukendrup is right. If it is **already** saved **Privately** to the cloud, then CTRL+S should just save it with no dialogs. (Same as pressing CTRL+S on a local file).

However, if the score has visibility set to **Public**, and presses CTRL+S, it's different because it's live and requires new audio to be uploaded. In that case, they need to re-upload new audio and there is a specific dialog informing of this first. (You can see it in the design). 

Just in case anyone is interested, we're putting in this friction for saving **Public** scores to encourage people not to save unfinished work privately.

Reply:
Okay, that's clear. I'd like to add two things (although they are a bit obvious):
- between saves, the user might have changed the visibility of the score using the score manager on MuseScore.com. So, when the user presses Ctrl+S, we should first download the latest score metadata to see if the score is _actually_ public, and act according to that, rather than to whatever information we have cached.
- To save time and energy, we don't need to do anything when the user presses Ctrl+S in a score where all changes have already been saved/uploaded successfully (i.e. score not "dirty"). In other words, do nothing when nothing needs to be done.

Reply:
I misunderstood this part of the design. Thank you for correcting me